internal/storage/filesystem.go

```diff
@@ -111,5 +112,20 @@ func (fs *filesystem) WriteFile(path string, contents []byte) error {
 		return err
 	}
 
-	return os.WriteFile(filepath.Join(dir, path), contents, 0600)
+	return write(filepath.Join(dir, path), contents)
+}
+
+func write(where string, data []byte) error {
+	fd, err := os.OpenFile(where, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0600)
+	if err != nil {
+		return fmt.Errorf("creating %s failed: %w", where, err)
```

Comment:
```suggestion
		return fmt.Errorf("opening %s failed: %w", where, err)
```

Reply:
Either way probably works. The flags will create `where` if it doesn't already exist.